docs: clarify .env loading precedence / behavior (#18765)
vitejs/vite: docs/guide/env-and-mode.md

> An env file for a specific mode (e.g. `.env.production`) will take higher priority than a generic one (e.g. `.env`).
> 
> ⚠️ When running with a mode specified, Vite will load _both_ `.env` and `.env.[mode]` files. If the same variable is defined in both files, the value in `.env.[mode]` will take precedence. Variables defined only in `.env` will also be passed into the environment unless explicitly overridden in `.env.[mode]`.

Reviewer: ```suggestion
The variables declared in an env file for a specific mode (e.g. `.env.production`) will take higher priority than the ones in a generic one (e.g. `.env`).
```
The fact that `.env` is always loaded is written above in the code block. Does changing the sentence to clarify that the priority is talking about the variables rather than the files make things clear?

Author: @sapphi-red 

Thanks for the review and the proposed update!

I agree that your wording is an improvement, but I still believe that Vite's behavior here warrants a clearer explanation or warning in the documentation.

The current behavior, where `.env` and .`env.local` are implicitly/magically loaded regardless of the specified mode, might feel unexpected or unconventional to many developers.

For example, developers coming from tools like `dotenv` in NodeJS or `python-dotenv` are accustomed to _explicitly_ loading the desired `.env` file without the additional inclusion of unrelated files.

While I understand that this implicit loading mimics the behavior of tools like Create React App (CRA), Vue CLI etc, I find it unintuitive and potentially problematic. Specifically, it can lead to surprises in production deployments where environment variables from .env (intended for local development) unintentionally make their way into the production environment.

I also understand that the current documentation mentions that the `.env` and `.env.local` files are "always loaded", but this detail can easily be overlooked or misunderstood. 

Including a clear and prominent note in the section on mode-specific .env files would help ensure that users fully understand this behavior.

How about something like:

> When running with a specific mode, Vite will always load `.env` and `.env.local` in addition to the mode-specific `.env.[mode]` file. Variables declared in mode-specific files will take precedence over those in generic files, but variables defined only in .env or .env.local will still be available in the environment.

Reviewer: > When running with a specific mode, Vite will always load `.env` and `.env.local` in addition to the mode-specific `.env.[mode]` file. Variables declared in mode-specific files will take precedence over those in generic files, but variables defined only in .env or .env.local will still be available in the environment.

Looks good to me. I think we can remove the `When running with a specific mode, ` as Vite always have a mode set (it's `development` in dev and `production` in build by default).

Author: > Looks good to me. I think we can remove the When running with a specific mode,  as Vite always have a mode set (it's development in dev and production in build by default).

Makes sense! Thanks for the feedback, I have applied your suggested changes.
